packages/core/rfcs/use-user-address-factory.md

```diff
@@ -0,0 +1,120 @@
+# UseUserAddressFactory with Commerce Tools composable use case
+
+## Motivation
+
+We need to extend useUserAddress type to show and modify user addresses.
+Also, some eCommerce platforms requires updating user entity with every address modification (for example: CommerceTools).
+
+## Composable interface
+```TS
+export interface UseUserAddress<ADDRESS, SEARCH_PARAMS, ADDRESS_OPTIONS = unknown> {
+  addresses: ComputedProperty<ADDRESS[]>;
+  shippingAddresses: ComputedProperty<ADDRESS[]>;
+  billingAddresses: ComputedProperty<ADDRESS[]>;
+  loadAddresses: (params?: SEARCH_PARAMS) => Promise<void>;
+  addAddress: (address: ADDRESS, options?: ADDRESS_OPTIONS) => Promise<void>;
+  updateAddress: (address: ADDRESS, options?: ADDRESS_OPTIONS) => Promise<void>;
+  deleteAddress: (address: ADDRESS, options?: ADDRESS_OPTIONS) => Promise<void>;
+  loading: ComputedProperty<boolean>;
+}
+```
+
+## Factory params interface
+```TS
+// factory params schema
+export type UseUserAddressFactoryParams<ADDRESS, SEARCH_PARAMS, ADDRESS_OPTIONS = unknown> = {
+  loadAddresses: (params?: SEARCH_PARAMS) => Promise<ADDRESS[]>;
+  getBillingAddresses: (addresses: ADDRESS[]) => ADDRESS[];
+  getShippingAddresses: (addresses: ADDRESS[]) => ADDRESS[];
+  addAddress: (address: ADDRESS, options?: ADDRESS_OPTIONS) => Promise<ADDRESS[]>;
+  updateAddress: (address: ADDRESS, options?: ADDRESS_OPTIONS) => Promise<ADDRESS[]>;
+  deleteAddress: (address: ADDRESS, options?: ADDRESS_OPTIONS) => Promise<ADDRESS[]>;
+}
+```
+
+##  Other changes requirements:
+We need to add a possibility to update a user
+```TS
+// FACTORY INTERFACE
+export interface UseUser
+<
+  USER,
+  UPDATE_USER_PARAMS
+> {
+  // ...leave all the stuff as they are now
+
+  // needed to add this:
+  setUser: (newUser: USER) => void;
```

Comment:
It shouldn't be a part of params, we have access to this object from the inside therefore we can just return this fn

Reply:
Please take a look at @andrzejewsky RFC